linetools/isgm/abssightline.py

```diff
@@ -0,0 +1,242 @@
+        """
+        # Coordinates
+        if chk_sep:
+            testcoord = bool(self.coord.separation(abscomp.coord) < tol)
+        else:
+            testcoord = True
+
+        # Combine
+        test = testcoord
+        # Append?
+        if test:
+            self._components.append(abscomp)
+        else:
+            warnings.warn("Failed add_component test")
+            print('Input absline with wrest={:g} does not match component rules. Not appending'.format(absline.wrest))
+            if not testcoord:
+                print("AbsComp coordinates do not match.  Best to set them")
+
+    def build_table(self):
+        """Generate an astropy QTable out of the components.
+        Default columns are z, Ion, flag_N, logN, sig_logN
+
+        Returns
+        -------
+        comp_tbl : Table
+        """
+        from linetools.abund.ions import ion_name
+        comp_tbl = Table()
+        if len(self._components) == 0:
+            return comp_tbl
+        # z
+        comp_tbl.add_column(Column([icomp.zcomp for icomp in self._components], name='z'))
+        # Ion
+        comp_tbl.add_column(Column([ion_name(icomp.Zion) for icomp in self._components], name='Ion'))
+        # Rest
+        for attrib in ['flag_N', 'logN', 'sig_logN']:
+            comp_tbl.add_column(Column([getattr(icomp, attrib)
+                                       for icomp in self._components], name=attrib))
+        # Return
+        return comp_tbl
+
+    def to_dict(self):
+        """ Write AbsSightline data to a dict that can be written with JSON
+        """
+        import datetime
+        import getpass
+        date = str(datetime.date.today().strftime('%Y-%b-%d'))
+        user = getpass.getuser()
+        # Generate the dict
+        outdict = dict(RA=self.coord.ra.value, DEC=self.coord.dec.value,
+                       CreationDate=date, user=user)
+        outdict['class'] = self.__class__.__name__
+        # Add other attributes
+        all_attr = self.__dict__
+        for key in all_attr:
+            if key in ['coord', '_components']:
+                continue
+            else:
+                outdict[key] = getattr(self, key)
+        # Components
+        outdict['components'] = {}
+        for component in self._components:
+            outdict['components'][component.name] = ltu.jsonify(component.to_dict())
+        # Polish
+        outdict = ltu.jsonify(outdict)
+        # Return
+        return outdict
+
+    def __repr__(self):
+        txt = '<{:s}: {:s} {:s}'.format(
+            self.__class__.__name__, self.coord.ra.to_string(unit=u.hour,sep=':', pad=True),
+                self.coord.dec.to_string(sep=':',pad=True,alwayssign=True))
+
+        # Type?
+        if self.em_type is not None:
+            txt = txt + ', emtype={:s}'.format(self.em_type)
+
+        # Finish
+        txt += '>'
+        return (txt)
+
+
+class GenericAbsSightline(AbsSightline):
+    """Class for Generic Absorption Sightline
```

Comment:
What is the idea of this Class? I think it should be a GenericSightline class, and then AbsSightline would be a child class of the generic one. But you may have another idea in mind...